Gems/PhysX/Code/Source/EditorPrismaticJointComponent.cpp

```diff
@@ -0,0 +1,217 @@
+                ;
+
+            if (auto* editContext = serializeContext->GetEditContext())
+            {
+                editContext
+                    ->Class<EditorPrismaticJointComponent>(
+                        "PhysX Prismatic Joint", "A dynamic joint that constrains a rigid body with linear limits along a single axis.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "PhysX")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(
+                        0,
+                        &EditorPrismaticJointComponent::m_linearLimit,
+                        "Linear Limit",
+                        "The limit of linear motion along the joint's axis.");
+            }
+        }
+    }
+
+    void EditorPrismaticJointComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("PhysicsJointService"));
+    }
+
+    void EditorPrismaticJointComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("TransformService"));
+        required.push_back(AZ_CRC_CE("PhysicsRigidBodyService"));
+    }
+
+    void EditorPrismaticJointComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("NonUniformScaleService"));
+    }
+
+    void EditorPrismaticJointComponent::Activate()
+    {
+        EditorJointComponent::Activate();
+
+        const AZ::EntityId& entityId = GetEntityId();
+
+        AzToolsFramework::EditorComponentSelectionRequestsBus::Handler::BusConnect(entityId);
+        AzToolsFramework::EditorComponentSelectionNotificationsBus::Handler::BusConnect(entityId);
+        PhysX::EditorJointRequestBus::Handler::BusConnect(AZ::EntityComponentIdPair(entityId, GetId()));
+    }
+
+    void EditorPrismaticJointComponent::Deactivate()
+    {
+        PhysX::EditorJointRequestBus::Handler::BusDisconnect();
+        AzToolsFramework::EditorComponentSelectionNotificationsBus::Handler::BusDisconnect();
+        AzToolsFramework::EditorComponentSelectionRequestsBus::Handler::BusDisconnect();
+        EditorJointComponent::Deactivate();
+    }
+
+    void EditorPrismaticJointComponent::BuildGameEntity(AZ::Entity* gameEntity)
+    {
+        m_config.m_followerEntity = GetEntityId(); // joint is always in the same entity as the follower body.
+        gameEntity->CreateComponent<PrismaticJointComponent>(
+            m_config.ToGameTimeConfig(),
+            m_config.ToGenericProperties(),
+            m_linearLimit.ToGameTimeConfig());
+    }
+
+    float EditorPrismaticJointComponent::GetLinearValue(const AZStd::string& parameterName)
```

Comment:
question: Could we use CRCs here instead instead of the string comparisons?

Reply:
(repeated elsewhere too)

Reply:
This is just the pattern in the rest of the joints. We could use AZ::Name or CRC like you suggest, but I think it would make sense to do that as a separate change for all the joints.

Reply:
Okay sounds like a good plan 👍